src/vs/workbench/api/node/extHostTerminalService.ts

```diff
@@ -23,6 +23,16 @@ export class ExtHostTerminalService extends BaseExtHostTerminalService {
 	public createTerminalFromOptions(options: vscode.TerminalOptions, internalOptions?: ITerminalInternalOptions): vscode.Terminal {
 		const terminal = new ExtHostTerminal(this._proxy, generateUuid(), options, options.name);
 		this._terminals.push(terminal);
+		if (options.location && typeof options.location === 'object' && 'parentTerminal' in options.location) {
```

Comment:
This could be removed if we take the `vscode.Terminal` approach as we only convert it to a `ExtHostTerminal` when we need to serialize to go to the other process